src/store.ts

```diff
@@ -113,41 +202,68 @@ export default class IndexedFormula extends Formula { // IN future - allow pass
     this.initPropertyActions(this.features)
   }
 
-  static get defaultGraphURI () {
+  /**
+   * Gets the URI of the default graph
+   */
+  static get defaultGraphURI(): string {
     return defaultGraphURI
   }
 
-  substitute (bindings) {
-    var statementsCopy = this.statements.map(function (ea) {
-      return ea.substitute(bindings)
+  /**
+   * Gets this graph with the bindings substituted
+   * @param bindings The bindings
+   */
+  //@ts-ignore different from signature in Formula
+  substitute(bindings: Bindings): IndexedFormula {
+    var statementsCopy = this.statements.map(function (ea: Quad) {
+      return (ea as Statement).substitute(bindings)
     })
     var y = new IndexedFormula()
     y.add(statementsCopy)
     return y
   }
 
-  addDataCallback(cb) {
+  /**
+   * Add a callback which will be triggered after a statement has been added to the store.
+   * @param cb
+   */
+  addDataCallback(cb: (q: Quad) => void): void {
     if (!this.dataCallbacks) {
       this.dataCallbacks = []
     }
     this.dataCallbacks.push(cb)
   }
 
-  applyPatch (patch, target, patchCallback) { // patchCallback(err)
+  /**
+   * Apply a set of statements to be deleted and to be inserted
+   *
+   * @param patch - The set of statements to be deleted and to be inserted
+   * @param target - The name of the document to patch
+   * @param patchCallback - Callback to be called when patching is complete
+   */
+  applyPatch(
+    patch: {
+        delete?: ReadonlyArray<Statement>,
+        patch?: ReadonlyArray<Statement>,
+        where?: any
+    },
+    target: TFNamedNode,
+    patchCallback: (errorString: string) => void
+  ): void {
     var targetKB = this
     var ds
-    var binding = null
+    var binding: Bindings | null = null
 
-    function doPatch (onDonePatch) {
+    function doPatch (onDonePatch: Function) {
```

Comment:
You could be a bit more specific on the Function, e.g. `() => void`

Reply:
True, fixed.